controllers/topology/desired_state.go

```diff
@@ -58,7 +60,17 @@ func (r *ClusterReconciler) computeDesiredState(_ context.Context, class *cluste
 	// InfrastructureCluster and the ControlPlane objects generated by the previous step.
 	desiredState.cluster = computeCluster(current, desiredState.infrastructureCluster, desiredState.controlPlane.object)
 
-	// TODO: implement generate desired state for machine deployments
+	// Compute the desired state of the MachineDeployment objects for the worker nodes.
+	if len(current.cluster.Spec.Topology.Workers.MachineDeployments) > 0 {
```

Comment:
Can we flip this condition to be `== 0` and return right away? 

Or maybe move it to a switch case?